src/lib/services/event-service.ts

```diff
@@ -35,4 +41,44 @@ export default class EventService {
             totalEvents,
         };
     }
+
+    async storeEvent(event: IBaseEvent): Promise<void> {
+        if (event.featureName && !event.tags) {
+            const tags = await this.featureTagStore.getAllTagsForFeature(
+                event.featureName,
+            );
+            return this.eventStore.store({ ...event, tags });
+        }
+        return this.eventStore.store(event);
+    }
```

Comment:
I think we should just
```suggestion
    async storeEvent(event: IBaseEvent): Promise<void> {
        return this.storeEvents([event]);
    }
```